src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -0,0 +1,932 @@
+            sequence are not taken into account for computing the loss.
+        question_positions (:obj:`torch.LongTensor` of shape :obj:`(batch_size, num_questions)`, `optional`):
+            The positions of all question tokens. If given, start_logits and end_logits will be of shape
+            :obj:`(batch_size, num_questions, sequence_length)`. If None, the first question token in each sequence in
+            the batch will be the only one for which start_logits and end_logits are calculated and they will be of
+            shape :obj:`(batch_size, sequence_length)`.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        question_positions_were_none = False
+        if question_positions is None:
+            if input_ids is not None:
+                question_position_for_each_example = torch.argmax(
+                    (torch.eq(input_ids, self.question_token_id)).int(), dim=-1
+                )
+            else:
+                question_position_for_each_example = torch.zeros(
+                    inputs_embeds.size(0), dtype=torch.long, layout=inputs_embeds.layout, device=inputs_embeds.device
+                )
+            question_positions = question_position_for_each_example.unsqueeze(-1)
+            question_positions_were_none = True
+
+        outputs = self.splinter(
+            input_ids,
+            attention_mask=attention_mask,
+            token_type_ids=token_type_ids,
+            position_ids=position_ids,
+            head_mask=head_mask,
+            inputs_embeds=inputs_embeds,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        sequence_output = outputs[0]
+        start_logits, end_logits = self.splinter_qass(sequence_output, question_positions)
+
+        if question_positions_were_none:
+            start_logits, end_logits = start_logits.squeeze(1), end_logits.squeeze(1)
+
+        if attention_mask is not None:
+            start_logits = start_logits + (1 - attention_mask) * -10000.0
+            end_logits = end_logits + (1 - attention_mask) * -10000.0
+
+        total_loss = None
+        if start_positions is not None and end_positions is not None:
+            # If we are on multi-GPU, split add a dimension
+            if len(start_positions.size()) > 1:
+                start_positions = start_positions.squeeze(-1)
+            if len(end_positions.size()) > 1:
+                end_positions = end_positions.squeeze(-1)
+            # sometimes the start/end positions are outside our model inputs, we ignore these terms
+            ignored_index = start_logits.size(1)
+            start_positions.clamp_(0, ignored_index)
+            end_positions.clamp_(0, ignored_index)
+
+            loss_fct = CrossEntropyLoss(ignore_index=ignored_index)
```

Comment:
Usually we don't overwrite the default `-100` with a unique `ignored_index`, but I think it's fine in this case as there is no padding on the targets that would have to be replaced with `-100`

Reply:
Ok, I'm not opposed to that